### ISSUE TITLE
[Bug]: Invalid or no prefix found error on network switch

### ISSUE THREAD

**Issue by kpav33** [MetaMask/metamask-extension#21040]
### Describe the bug

On some dapps, when trying to switch a network programmatically, the Invalid or no prefix found bug is shown in MetaMask and the network doesn't switch successfully. In my case I was trying to switch network by using the wagmi library, but there have been reports by other developers with this same bug, while using the ethereum.request directly. See this thread https://community.metamask.io/t/metamask-invalid-or-no-prefix-found-error-on-network-switch/26987 for more information.

So briefly, trying to switch network programmatically, results in the above mentioned error, but switching networks manually in MetaMask, works OK. This seems to be an issue with the newest version of MetaMask (v11.0.0), after I downgraded MetaMask to v10.35.1, the error no longer happened. However this only seems to affect some dapps and it sometimes has a tendency to simply resolve on its own, after a couple days.

### Expected behavior

The network should switch, without any issues.

### Steps to reproduce

1. Go to https://luxy.io/ (this is not the website that I have been working on, but from another developer with the same issue, since mine aren't deployed to production yet).
2. First login into your MetaMask extension (check that your MetaMask is the latest 11.0.0 version) and switch to one of the non-supported networks for the https://luxy.io/ website, so that after connecting you will get the switch network button (I connected to Goerli, but any testnet would work).
3. Click on the Connect button in the top right and click on the Sign button.
4. Now you should see a Network not supported button in the top right, if you don't see it, switch to some other network manually in your MetaMask. Click on the Network not supported button and select the Ethereum network button in the opened modal.
5. After clicking on the Ethereum button, MetaMask should try to switch the network and this should result in an error.

### Error messages or log output

```shell
Message: Invalid or no prefix found for "https://luxy.io"

Code: Error

Stack:
Error: Invalid or no prefix found for "https://luxy.io"
  at n.getSnapPrefix (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-1.js:1:75297)
  at F (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-7.js:10:158232)
  at t.getSnapName (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/common-7.js:10:158399)
  at x (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-9.js:1:50677)
  at va (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-3.js:1:60111)
  at Mu (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-3.js:1:105038)
  at As (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-3.js:1:90901)
  at Rs (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-3.js:1:90826)
  at xs (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-3.js:1:88166)
  at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-3.js:1:46516
  at n.unstable_runWithPriority (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-3.js:1:514297)
  at Xo (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-3.js:1:46225)
  at ti (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-3.js:1:46461)
  at ei (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-3.js:1:46396)
  at Es (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-3.js:1:88543)
  at dl (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-3.js:1:113783)
  at render (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-3.js:1:115706)
  at chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/ui-8.js:1:199025
```

### Version

11.0.0

### Build type

None

### Browser

Chrome, Firefox

### Operating system

Windows

**Comment by anaamolnar** [comment]
Hello, @kpav33. Thanks for taking the time to write this report! I will pass this on to the team.

**Comment by manazpr** [comment]
I have had the same issue since last couple of weeks, and many people are now discovering the issue after updating to the latest version. Hope it's fixed soon.😔

**Comment by lucky7323** [comment]
If there is no title's content of `index.html`, the error occurs. 

```
<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <meta name="title" content="FILL_HERE!" />
...
```

**Comment by kpav33** [comment]
@jacobpake Yes, I am also noticing this error happening again, so the issue doesn't seems to be completely fixed yet.
@lucky7323 On the apps that I had this issue with, I have the title content set and I still get the error, so this might work for some people, but didn't work for me.
@anaamolnar @GuillaumeRx Can you guys look into this issue again, since we are still experiencing problems with it?

**Comment by kpav33** [comment]
Same here, also using Firefox and I thought the fix was in 11.2.0. If I update to 11.3.0 the bug goes away.
